src/SpectatorView.Unity/Assets/SpectatorView.Editor/Scripts/CompositorWindow.cs

```diff
@@ -128,27 +130,47 @@ private void CompositeGUI()
                     RenderTitle(title, Color.green);
                 }
 
-                EditorGUILayout.BeginHorizontal("Box");
+                EditorGUILayout.BeginVertical("Box");
                 {
-                    string[] compositionOptions = new string[] { "Final composite", "Intermediate textures" };
-                    GUIContent renderingModeLabel = new GUIContent("Preview display", "Choose between displaying the composited video texture or seeing intermediate textures displayed in 4 sections (bottom left: input video, top left: opaque hologram, top right: hologram alpha mask, bottom right: hologram alpha-blended onto video)");
-                    textureRenderMode = EditorGUILayout.Popup(renderingModeLabel, textureRenderMode, compositionOptions);
-                    if (compositionManager != null && compositionManager.TextureManager != null)
-                    {
-                        compositionManager.TextureManager.IsQuadrantVideoFrameNeededForPreviewing = textureRenderMode == (int)VideoRecordingFrameLayout.Quad;
-                    }
-                    FullScreenCompositorWindow fullscreenWindow = FullScreenCompositorWindow.TryGetWindow();
-                    if (fullscreenWindow != null)
+                    EditorGUILayout.Space();
+                    EditorGUILayout.BeginHorizontal();
                     {
-                        fullscreenWindow.TextureRenderMode = textureRenderMode;
+                        if (compositionManager != null)
+                        {
+                            GUI.enabled = compositionManager == null || !compositionManager.IsVideoFrameProviderInitialized;
+                            GUIContent label = new GUIContent("Video source", "The video capture card you want to use as input for compositing.");
+                            compositionManager.CaptureDevice = (FrameProviderDeviceType)EditorGUILayout.Popup(label, ((int)compositionManager.CaptureDevice), Enum.GetValues(typeof(FrameProviderDeviceType)).Cast<FrameProviderDeviceType>().Where(provider => compositionManager.IsFrameProviderSupported(provider) || (provider == FrameProviderDeviceType.None)).Select(x => x.ToString()).ToArray());
```

Comment:
Could you break this out into several lines variables; hard to read.

Also, I like to stack my linq-esk calls:
SomeList()
  .Where()
  .Select()
  .ToArray()

Reply:
yeah that would help, let me fix things.